tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,263 @@
+    assert custom_host.run("dnf install leapp-upgrade -y").status == 0
+
+
+def fix_inhibitors(custom_host, source_rhel):
+    """Fixing inhibitors to avoid hard stop of Leapp tool execution"""
+    if '8' in source_rhel:
+        # 1. Firewalld Configuration AllowZoneDrifting Is Unsupported
+        custom_host.run(
+            'sed -i "s/^AllowZoneDrifting=.*/AllowZoneDrifting=no/" /etc/firewalld/firewalld.conf'
+        )
+        # 2. Newest installed kernel not in use
+        if custom_host.run('needs-restarting -r').status == 1:
+            custom_host.power_control(state='reboot', ensure=True)
+    else:
+        # placeholder for source_rhel - 7
+        pass
+
+
+@pytest.fixture(scope="module")
+def setup_env(module_target_sat, module_sca_manifest_org):
+    """Creating essential things and returning in form of dictiory for use"""
+    # 1. Import a subscription manifest
+    lc_env = module_target_sat.api.LifecycleEnvironment(
+        organization=module_sca_manifest_org
+    ).create()
+    c_view = module_target_sat.api.ContentView(organization=module_sca_manifest_org).create()
+    return {
+        'organization': module_sca_manifest_org,
+        'satellite': module_target_sat,
+        'environment': lc_env,
+        'contentview': c_view,
+    }
+
+
+@pytest.mark.no_containers
+@pytest.mark.parametrize(
+    'custom_host',
+    [
+        {'deploy_rhel_version': '8.8'},
+    ],
+    ids=['RHEL8.8'],
+    indirect=True,
+)
+@pytest.mark.parametrize('target_rhel', ['rhel-9'])
+def test_upgrade_rhel8_to_rhel9(
+    setup_env,
+    custom_host,
+    target_rhel,
+):
+    """Test to upgrade RHEL host to next major RHEL Realse with Leapp Preupgrade and Leapp Upgrade
+    Job templates
+
+    :id: 7add9b32-a13a-4e65-973c-cd64326b332a
+
+    :Steps:
+        1. Import a subscription manifest and enable, sync source & target repositories
+        2. Create CV, add repositories, create lce, ak, etc.
+        3. Register contenthost using ak
+        4. Varify target rhel repositories are enable on Satellite
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+
+    :expectedresults:
+        1. Update RHEL OS major version to another major version
+
+    """
+    # rhel_contenthost = custom_host
+    organization = setup_env['organization']
+    target_sat = setup_env['satellite']
+    lc_env = setup_env['environment']
+    c_view = setup_env['contentview']
+
+    # Enable rhel8/rhel9 bos, aps repository and add in content view
+    all_repos = []
+    for rh_repo_key in CUSTOM_REPOS.keys():
+        prod = rh_repo_key.split('_')[0]
+        repo_id = target_sat.api_factory.enable_rhrepo_and_fetchid(
+            basearch=custom_host.arch,
+            org_id=organization.id,
+            product=PRDS[prod],
+            repo=CUSTOM_REPOS[rh_repo_key]['name'],
+            reposet=CUSTOM_REPOSET[rh_repo_key],
```

Comment:
with merged `CUSTOM_REPOSET` into `CUSTOM_REPOS` you need to write here 
```
reposet=CUSTOM_REPOS[rh_repo_key]['reposet'],
```

Reply:
Thank you for the suggestion and related changes,  I will take this in upcoming PR.